Source/WebCore/animation/CSSPropertyAnimation.cpp

```diff
@@ -374,7 +374,9 @@ static inline TextUnderlineOffset blendFunc(const TextUnderlineOffset& from, con
 static inline TextDecorationThickness blendFunc(const TextDecorationThickness& from, const TextDecorationThickness& to, const CSSPropertyBlendingContext& context)
 {
     if (from.isLength() && to.isLength())
-        return TextDecorationThickness::createWithLength(blendFunc(from.lengthValue(), to.lengthValue(), context));
+        return TextDecorationThickness::createWithLength(blendFunc(from.value(), to.value(), context));
+    if (from.isPercentage() && to.isPercentage())
+        return TextDecorationThickness::createWithPercentage(blendFunc(from.value(), to.value(), context));
```

Comment:
Looks good with the current `<length> | <percentage>` syntax. If we change to `<length-percentage>` as mentioned in https://github.com/WebKit/WebKit/pull/19085/files#r1360007422, then this code will need to change as well to pass the flag allowing blending between `<length>` and `<percentage>` values.

Reply:
The spec seems to be likely corrected. I adopted `<length-percentage>` in the patch.